src/api/environment.cc

```diff
@@ -45,6 +47,37 @@ static bool ShouldAbortOnUncaughtException(Isolate* isolate) {
          !env->inside_should_not_abort_on_uncaught_scope();
 }
 
+static MaybeLocal<Value> PrepareStackTraceCallback(Local<Context> context,
+                                      Local<Value> exception,
+                                      Local<Array> trace) {
+  Environment* env = Environment::GetCurrent(context);
+  if (env == nullptr) {
+    MaybeLocal<String> s = exception->ToString(context);
+    return s.IsEmpty() ?
+      MaybeLocal<Value>() :
+      MaybeLocal<Value>(s.ToLocalChecked());
+  }
+  Local<Function> prepare = env->prepare_stack_trace_callback();
+  if (prepare.IsEmpty()) {
+    MaybeLocal<String> s = exception->ToString(context);
+    return s.IsEmpty() ?
+      MaybeLocal<Value>() :
+      MaybeLocal<Value>(s.ToLocalChecked());
+  }
+  Local<Value> args[] = {
+      context->Global(),
+      exception,
+      trace,
+  };
+  TryCatchScope try_catch(env);
```

Comment:
Why do we need `TryCatchScope` and `Rethrow` here if it is not fatal? I think we can just return the empty `MaybeLocal` if the function throws an error?

Reply:
See https://crbug.com/v8/7848#c26 for context. The TryCatch + Rethrow is required by V8 due to details around exception handling there. For C++ callbacks, V8 expects a scheduled exception (which is what Rethrow gives us). Just returning the empty MaybeLocal would leave us with a pending exception.

Reply:
Thanks for the pointer. Can we put a comment here referencing the link?